docs/styles/docs.scss

```diff
@@ -1,66 +1,116 @@
-.docs__container {
+@use 'tools' as *;
+
+.docs {
   --nav-width: 200px;
   --site-gap: 40px;
-  display: grid;
-  grid-template-columns: 1fr;
-  max-width: calc(var(--content-max-width) + var(--nav-width));
-  width: 100%;
-  align-items: flex-start;
-  margin: auto;
-  padding: 0 20px;
-  gap: var(--site-gap);
+  --site-max-width: calc(var(--content-max-width) + var(--nav-width));
+
+  &__container {
+    display: grid;
+    grid-template-columns: 1fr;
+    max-width: calc(var(--content-max-width) + var(--nav-width));
+    width: 100%;
+    align-items: flex-start;
+    margin: auto;
+    padding: 0 20px;
+    gap: var(--site-gap);
+  
+    @media (min-width: breakpoint(tablet)) {
+      grid-template-columns: [side-start] var(--nav-width) [side-end content-start] 1fr [content-end];
+    }
+    
+    &__content {
+      overflow: auto;
+      max-width: 100%;
+      margin: 0;
+      padding: 0;
+  
+      > h1 {
+        margin-block-start: var(--site-gap);
+      }
+  
+      @media (min-width: breakpoint(tablet)) {
+        grid-column: content-start / content-end;
+      }
+  
+      @include gradient-links;
+    }
+  }
+
+  &__primary-nav {
+    --max-width: var(--site-max-width);
+    height: 4rem;
+    display: grid;
+    align-items: center;
+
+    &__content {
+      width: 100%;
+      align-items: center;
+      margin: auto;
+      padding: 0 1rem;
+    }
+
+    a:not(.docs__home-link) {
+      @include gradient-link;
+      color: white;
+      text-decoration: none;
+    }
+  }
 
-  @media (min-width: $tablet) {
-    grid-template-columns: [side-start] var(--nav-width) [side-end content-start] 1fr [content-end];
+  &__home-link, &__home-link:visited {
+    color: white;
+    text-decoration: none;
+    font-weight: bold;
+    font-size: 32px;
+    margin-right: auto;
   }
 
-  .docs__nav {
+
+  &__side-nav {
     list-style: none;
     padding: 16px 0;
     background: var(--grey-100);
     border-radius: 10px;
     position: sticky;
     top: var(--site-gap);
-
-    @media (min-width: $tablet) {
+  
+    @include gradient-links;
+  
+    @media (min-width: breakpoint(tablet)) {
       margin-block-start: var(--site-gap);
       grid-column: side-start / side-end;
     }
-
+  
     li {
       margin: 0;
     }
-
-    li > a {
+  
+    a, a:visited {
       --bg: var(--grey-100);
       display: block;
       padding: 10px 20px;
```

Comment:
the 20px/10px are seen in a few places in this file. I wonder if it makes sense to establish a custom property (for example, `--site-spacing`) instead of setting 20px explicitly again. That only makes sense if all of these 20px are actually _related_.

The same question is specifically important here. Is it "10px", or is ist "half of 20px", or "half of general spacing"?. If you express this with a calculation, you are defining relationships instead of values.

Or are they even related to `--site-gap`? In that case they can be derived from it with `calc()` as well. It's not about DRY its about relationship.

```css
padding: calc(var(--site-spacing)/2) var(--site-spacing);
```
or
```css
--int-padding: 10px; /* whatever naming makes sense to you */
padding: var(--int-padding) calc(var(--int-padding) * 2);
```

Reply:
Hm, I think the more general problem is using loose `px` values everywhere. I'd prefer replacing all of these with `rem`. Assuming a font size of `16px`, this would allow pretty simple calculations to get multiples of 8:
- `0.5rem - 8px`
- `1rem - 16px`
...

I think this pattern of using half-steps of `rem` should catch on throughout the codebase

Reply:
Right, in that case the relationship is to the documents font-size, not a general spacing. All of it makes sense - it depends on how you personally model it when you decided on this value.

Reply:
Yeah for font size in particular, we _do_ have variables for `--h1`, `--h2`, `--h3`, etc. This particular heading was juuuuuust between an h3 and an h4 though, so I compromised 😕